waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/tunnelling/TunnelingMetaStoreClientFactory.java

```diff
@@ -41,55 +38,65 @@
   @VisibleForTesting
   final MethodChecker METHOD_CHECKER = new MetastoreClientMethodChecker();
 
-  private final TunnelableFactorySupplier tunnelableFactorySupplier;
+  private HiveConf localHiveConf;
```

Comment:
Don't store state in Factories. remove please keep that scoped to the `newInstance`

Reply:
If I remove this, then I won't be able to test if the local hive conf gets the correct parameters.
Apparently there was an issue before where the code was changed, and instead of getting local port and local host, `localHiveConf` got other parameters. Because that happened before, I thought it would be a good idea to have a test that prevents that from happening again.

Reply:
If I remove this, then I won't be able to test if the local hive conf gets the correct parameters.
Apparently there was an issue before where the code was changed, and instead of getting local port and local host, `localHiveConf` got other parameters. Because that happened before, I thought it would be a good idea to have a test that prevents that from happening again.